### PR TITLE
Remove NDStore Residue

### LINKED ISSUE #2878: refactor backend: remove NDStore residue

### PR DESCRIPTION
**Pull request by fm3**
### URL of deployed dev instance (used for testing):
- https://removendstoreresidue.webknossos.xyz

### Steps to test:
- datasets should still get reported
- tracings should still be created/downloaded/uploaded

### Issues:
- fixes #2878 

------
- ~~[ ] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)~~
- [x] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable
- [x] added PR number in evolution
- ~~[ ] Needs datastore update after deployment~~
- [x] Ready for review

### CODE REVIEW

File: conf/evolutions/reversions/022-add-foreign-datastore.sql [hunk 1]
```diff
@@ -2,4 +2,6 @@ START TRANSACTION;
 DROP VIEW webknossos.datastores_;
 ALTER TABLE webknossos.datastores DROP COLUMN isForeign;
 CREATE VIEW webknossos.dataStores_ AS SELECT * FROM webknossos.dataStores WHERE NOT isDeleted;
+
+UPDATE webknossos.releaseInformation set schemaVersion = 21;
```
Comment (jstriebel) [inline conf/evolutions/reversions/022-add-foreign-datastore.sql]: probably slipped in there

File: conf/evolutions/reversions/023-drop-datastore-type.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+START TRANSACTION;
+
+DROP VIEW webknossos.datastores_;
+ALTER TABLE webknossos.datastores add typ webknossos.DATASTORE_TYPE NOT NULL DEFAULT 'webknossos-store',
+CREATE VIEW webknossos.dataStores_ AS SELECT * FROM webknossos.dataStores WHERE NOT isDeleted;
+
+UPDATE webknossos.releaseInformation set schemaVersion = 22;
```
Comment (jstriebel) [inline conf/evolutions/reversions/023-drop-datastore-type.sql]: this is now evolution `23`

Comment (fm3) [reply conf/evolutions/reversions/023-drop-datastore-type.sql]: so the reversion has to set the schemaVersion back to 22 :)

Comment (jstriebel) [reply conf/evolutions/reversions/023-drop-datastore-type.sql]: I mixed it with the actual evolutions, didn't see it's the reversion, sorry

File: conf/evolutions/reversions/022-add-foreign-datastore.sql (hunk 1 above)
Comment (fm3) [reply conf/evolutions/reversions/022-add-foreign-datastore.sql]: so the reversion has to set it back to 21. (this was missing in #3080 )

File: conf/evolutions/reversions/023-drop-datastore-type.sql (hunk 2 above)
Comment (fm3) [reply conf/evolutions/reversions/023-drop-datastore-type.sql]: It’s very easy to miss, maybe the naming scheme should be different or something. Either way, thanks for the review :)

**Pull request merged**